core/iwasm/libraries/lib-socket/src/wasi/wasi_socket_ext.c

```diff
@@ -662,6 +663,22 @@ get_sol_socket_option(int sockfd, int optname, void *__restrict optval,
             error = __wasi_sock_get_broadcast(sockfd, (bool *)optval);
             HANDLE_ERROR(error);
             return error;
+        case SO_TYPE:
+            assert(*optlen == sizeof(int));
```

Comment:
Not introduced by this PR. However, regarding `__STD_TYPE __U32_TYPE __socklen_t`(in *bits/types.h*), should we be concerned that `sizeof(int)` might not be 32 bits?

Reply:
the value for SO_TYPE (eg. SOCK_STREAM) is int, not socklen_t.

Reply:
`socklen_t *__restrict optlen`. I am referring to the type of `optlen` that the assertion concerns.

Reply:
the assertion is about the value *optlen, not the type of it.
the type doesn't matter as far as it can represent sizeof(int).

Reply:
right. right.